### PR TITLE
CSHARP-3180: Use server error codes in place of error messages when possible.

### PR DESCRIPTION
**Pull request by DmitryLukyanov**
EG: https://evergreen.mongodb.com/version/5f616d5d7742ae771d51d0ed

### CODE REVIEW

**Comment by DmitryLukyanov** [comment]
I also found two more cases: 
1. https://github.com/mongodb/mongo-csharp-driver/blob/master/src/MongoDB.Driver.Legacy/MongoCollection.cs#L718
2. https://github.com/mongodb/mongo-csharp-driver/blob/master/src/MongoDB.Driver.Legacy/MongoCollection.cs#L782
but since it's a legacy, I think we can ignore it.

File: src/MongoDB.Driver.Core/ServerErrorCode.cs [hunk 1]
```diff
@@ -44,6 +44,7 @@ internal enum ServerErrorCode
         UnknownReplWriteConcern = 79,
         UnsatisfiableWriteConcern = 100,
         WriteConcernFailed = 64,
-        MaxTimeMSExpired = 50
+        MaxTimeMSExpired = 50,
+        NamespaceNotFound = 26
```
Comment (rstam) [inline src/MongoDB.Driver.Core/ServerErrorCode.cs]: Looks like we were trying to keep these in alphabetical order.

**Review by rstam — CHANGES_REQUESTED** [review]
Minor change requested.

File: src/MongoDB.Driver.Core/ServerErrorCode.cs (hunk 1 above)
Comment (DmitryLukyanov) [reply src/MongoDB.Driver.Core/ServerErrorCode.cs]: done

**Review by rstam — APPROVED** [review]
LGTM

**Review by MikalaiMazurenka — APPROVED** [review]
LGTM!

**Pull request closed without merging**